File: software/examples/csharp/ExampleInterrupt.cs
using System;
using Tinkerforge;

class Example
{
	private static string HOST = "localhost";
	private static int PORT = 4223;
	private static string UID = "XYZ"; // Change XYZ to the UID of your IO-16 Bricklet

	// Callback function for interrupt callback
	static void InterruptCB(BrickletIO16 sender, char port, byte interruptMask,
	                        byte valueMask)
	{
		Console.WriteLine("Port: " + port);
		Console.WriteLine("Interrupt Mask: " + Convert.ToString(interruptMask, 2));
		Console.WriteLine("Value Mask: " + Convert.ToString(valueMask, 2));
		Console.WriteLine("");
	}

	static void Main()
	{
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletIO16 io = new BrickletIO16(UID, ipcon); // Create device object

		ipcon.Connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Register interrupt callback to function InterruptCB
		io.InterruptCallback += InterruptCB;

		// Enable interrupt on pin 2 of port A
		io.SetPortInterrupt('a', 1 << 2);

		Console.WriteLine("Press enter to exit");
		Console.ReadLine();
		ipcon.Disconnect();
	}
}

File: software/examples/rust/example_interrupt.rs
use std::{error::Error, io, thread};
use tinkerforge::{io16_bricklet::*, ip_connection::IpConnection};

const HOST: &str = "localhost";
const PORT: u16 = 4223;
const UID: &str = "XYZ"; // Change XYZ to the UID of your IO-16 Bricklet.

fn main() -> Result<(), Box<dyn Error>> {
    let ipcon = IpConnection::new(); // Create IP connection.
    let io = Io16Bricklet::new(UID, &ipcon); // Create device object.

    ipcon.connect((HOST, PORT)).recv()??; // Connect to brickd.
                                          // Don't use device before ipcon is connected.

    // Create receiver for interrupt events.
    let interrupt_receiver = io.get_interrupt_receiver();

    // Spawn thread to handle received events. This thread ends when the `io` object
    // is dropped, so there is no need for manual cleanup.
    thread::spawn(move || {
        for interrupt in interrupt_receiver {
            println!("Port: {}", interrupt.port);
            println!("Interrupt Mask: {:b}", interrupt.interrupt_mask);
            println!("Value Mask: {:b}", interrupt.value_mask);
            println!();
        }
    });

    // Enable interrupt on pin 2 of port A
    io.set_port_interrupt('a', 1 << 2);

    println!("Press enter to exit.");
    let mut _input = String::new();
    io::stdin().read_line(&mut _input)?;
    ipcon.disconnect();
    Ok(())
}
